tests/unit/moduleapi/moduleconfigs.tcl

```diff
@@ -93,11 +123,30 @@ start_server {tags {"modules"}} {
         assert_equal [r config get moduleconfigs.enum] "moduleconfigs.enum one"
         assert_equal [r config get moduleconfigs.flags] "moduleconfigs.flags {one two}"
         assert_equal [r config get moduleconfigs.numeric] "moduleconfigs.numeric -1"
+        
+        # Check un-prefixed and aliased configuration
+        assert_equal [r config get unprefix-bool] "unprefix-bool yes"
+        assert_equal [r config get unprefix-bool-alias] "unprefix-bool-alias yes"
+        assert_equal [r config get unprefix.numeric] "unprefix.numeric -1"
+        assert_equal [r config get unprefix.numeric-alias] "unprefix.numeric-alias -1"
+        assert_equal [r config get unprefix-string] "unprefix-string {secret unprefix}"
+        assert_equal [r config get unprefix.string-alias] "unprefix.string-alias {secret unprefix}"
+        assert_equal [r config get unprefix-enum] "unprefix-enum one"
+        assert_equal [r config get unprefix-enum-alias] "unprefix-enum-alias one"
+                
+                
         r module unload moduleconfigs
     }
 
     test {test loadex functionality} {
-        r module loadex $testmodule CONFIG moduleconfigs.mutable_bool no CONFIG moduleconfigs.immutable_bool yes CONFIG moduleconfigs.memory_numeric 2mb CONFIG moduleconfigs.string tclortickle
+        r module loadex $testmodule CONFIG moduleconfigs.mutable_bool no \
+                                    CONFIG moduleconfigs.immutable_bool yes \
```

Comment:
Hello @moticless, If I pass the same config name or the same config name alias, multiple times, the last value is taken. It's OK.
But If I pass the name, and the alias, an error is reported. It seems to be a bug, it isn't?

<details>
<summary>Test details</summary>

```sh
127.0.0.1:6379> module loadex /tmp/moduleconfigs.so CONFIG unprefix.numeric-alias 123 CONFIG unprefix.numeric-alias 989
OK
127.0.0.1:6379> config get unprefix.numeric*
1) "unprefix.numeric"
2) "989"
3) "unprefix.numeric-alias"
4) "989"
127.0.0.1:6379> module unload moduleconfigs
OK
127.0.0.1:6379> module loadex /tmp/moduleconfigs.so CONFIG unprefix.numeric-alias 123 CONFIG unprefix.numeric 989
(error) ERR Error loading the extension. Please check the server logs.
127.0.0.1:6379> module loadex /tmp/moduleconfigs.so CONFIG unprefix.numeric 123 CONFIG unprefix.numeric 666
OK
127.0.0.1:6379> config get unprefix.numeric*
1) "unprefix.numeric"
2) "666"
3) "unprefix.numeric-alias"
4) "666"
```

</details>

Reply:
During "initial" scan, in case of duplication of same name, it is trivially resolved seamlessly by multiple assignment of the same name into same entry in dict `module_configs_queue` and keeping last value. 

In case of name and alias duplication, it is not possible. During scan we don't know yet module's name-alias pairs and we keep each as a distinct name in dict `module_configs_queue`. Later on, on module load, we cannot resolve which one was set last (Under the wood, `loadModuleConfigs()` iterates configs of a given module, map to the first one to encounter in dict `module_configs_queue` and the other "alias" in `loadModuleConfigs()` will be leftover , and in turn will fail the load).

I think we can live with that and even might claim it is a protection mechanism against "alias" mistakes. @MeirShpilraien , @oranagra WDYT?

Reply:
are we talking of modules registering duplicate configs?
the second one should be rejected IMO.

or about passing a user passing the same config (or alias of a config) twice?
i think that should be allowed and the last writer wins (in the sense that the alias is not a different config, but rather another name for the same variable).
but either way, let's keep it consistent with how redis behaves with the built in configs and their aliases.

Reply:
Ok. Please review my last commit to verify. (Given <Name,Alias> pair, LOADEX ok: "N+N" or "A+A". fails: "N+A")